MoreLinq/IntersectBy.cs

```diff
@@ -146,10 +146,10 @@ partial class MoreEnumerable {
             IEqualityComparer<TKey> keyComparer) {
 
             var keys = new HashSet<TKey>(second, keyComparer);
-            foreach (var item in first) {
-                var k = keySelector(item);
+            foreach (var element in first) {
+                var k = keySelector(element);
                 if (keys.Contains(k)) {
-                    yield return item;
+                    yield return element;
                     keys.Remove(k);
```

Comment:
[HashSet.Remove also reports success](https://msdn.microsoft.com/en-us/library/bb342694(v=vs.110).aspx). I know, not in your patch, but if you are already modifying...